lib/disposable.dart

```diff
@@ -181,4 +193,10 @@ abstract class Disposable implements _Disposable {
     _didDispose.complete();
     return null;
   }
+
+  void _throwIfNull(dynamic subscription, String name) {
```

Comment:
I was thinking something like this:
```dart
void doSomething(AwesomeObject obj) {
     // do some expensive things.
     obj.callMethod();
}
```
That could be rewritten to avoid unnecessarily doing anything expensive:
```dart
void doSomething(AwesomeObject obj) {
     throwIfNull(obj, 'doSomething');
     // do some expensive things.
     obj.callMethod();
}
```